File: java/src/main/java/falcon/api/Logger.java
package falcon.api;

import falcon.api.internal.FalconAbi;
import falcon.api.internal.PluginContext;

public final class Logger {
    private final PluginContext mContext;

    Logger(PluginContext context) {
        mContext = context;
    }

    public void info(String message) {
        mContext.log(FalconAbi.LOG_INFO, message);
    }

    public void warning(String message) {
        mContext.log(FalconAbi.LOG_WARNING, message);
    }

    public void error(String message) {
        mContext.log(FalconAbi.LOG_ERROR, message);
    }

    public void error(String message, Throwable throwable) {
        mContext.report(message, throwable);
    }
}

File: dotnet/Falcon.PluginAPI/Logger.cs
using Falcon.Interop;

namespace Falcon
{
    public sealed class Logger
    {
        private readonly nint _plugin;

        internal Logger(nint plugin)
        {
            _plugin = plugin;
        }

        public void Info(string message)
        {
            NativeApi.Log(_plugin, FalconConstants.LogInfo, message);
        }

        public void Warning(string message)
        {
            NativeApi.Log(_plugin, FalconConstants.LogWarning, message);
        }

        public void Error(string message)
        {
            NativeApi.Log(_plugin, FalconConstants.LogError, message);
        }
    }
}
